### PR TITLE
[JENKINS-68732] Filter box for dropdown lists wipes search input after collapsing and reopening

### PR DESCRIPTION
**Pull request by benebsiny**
See [JENKINS-68732](https://issues.jenkins.io/browse/JENKINS-68732).

Dropdown

The reason why the filter box doesn't work as expected is because the second parameter passed to [_applyFilterKeyword](https://github.com/benebsiny/jenkins/blob/06886ecff9a0fd352c95cd502078d3b39617355f/core/src/main/resources/lib/form/filter-menu-button/filter-menu-button.js#L42) is incorrect.

By [Line 43](https://github.com/benebsiny/jenkins/blob/06886ecff9a0fd352c95cd502078d3b39617355f/core/src/main/resources/lib/form/filter-menu-button/filter-menu-button.js#L43), we can know that the `filterInput` should be an `Event` object. However, in [Line 15](https://github.com/benebsiny/jenkins/blob/06886ecff9a0fd352c95cd502078d3b39617355f/core/src/main/resources/lib/form/filter-menu-button/filter-menu-button.js#L15), an `Element` object is passed to `_applyFilterKeyword`, which lead to the occurrence of the JavaScript error when we open the dropdown lists (It's amazing that no error message shown in F12 console).

Some functions are also affected, for example, previous search results retain after reopening the box even if the search pattern is cleared, and the [MIN_NUM_OPTIONS](https://github.com/benebsiny/jenkins/blob/06886ecff9a0fd352c95cd502078d3b39617355f/core/src/main/resources/lib/form/filter-menu-button/filter-menu-button.js#L2),  the number that determine whether to show the filter box, doesn't work.

In order to fix this problem, the second parameter is changed to accept the `Element` object, and the function callers change to pass the `Element` object.

Besides, according to the JIRA issue description, I retain the previous search pattern in the filter box. See demo above.

### Proposed changelog entries

* Retain the search pattern and results across collapsing and reopening the dropdown list.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since="TODO")` or `@Deprecated(since="TODO", forRemoval=true)` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6709"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
